openlibrary/plugins/upstream/forms.py

```diff
@@ -95,12 +97,18 @@ class RegisterForm(Form):
             help=_("Letters and numbers only please, and at least 3 characters."),
             autocapitalize="off",
             validators=[vlogin, username_validator],
+            pattern="[a-zA-Z0-9]{3,20}",
```

Comment:
Let's use the same regex as the official validator here (this is actually subtly different!); maybe something like this? Not sure what the best way to convert it to a string is!

```suggestion
            pattern=str(vlogin.rexp),
```